packages/admin/src/pages/build-application/[applicationId]/[sectionId]/question-type/index.page.tsx

```diff
@@ -51,123 +68,104 @@ export const getServerSideProps: GetServerSideProps = async ({
-  let responseType;
-  try {
-    applicationFormSummary = await getApplicationFormSummary(
-      applicationId,
-      sessionCookie
-    );
-    responseType = await getValueFromSession(
+    const questionSummary = (await getSummaryFromSession(
       'newQuestion',
-      'responseType',
       sessionCookie
-    );
-  } catch (err) {
-    // If we can't fetch the section name from the application form summary, redirect to the service error page
-    return questionErrorPageRedirect(applicationId);
-  }
+    )) as QuestionSummary;
+    const { optional, ...restOfQuestionSummary } = questionSummary;
+    const maxWords =
+      body.responseType === ResponseType.ShortAnswer
+        ? SHORT_QUESTION_WORD_LIMIT
+        : undefined;
 
-  let selectedRadio = '';
-  switch (responseType) {
-    case ResponseType.Dropdown: {
-      selectedRadio = 'Multiple choice';
-      break;
-    }
-    case ResponseType.MultipleSelection: {
-      selectedRadio = 'Multiple select';
-      break;
+    await postQuestion(sessionId, applicationId, sectionId, {
+      ...restOfQuestionSummary,
+      ...props,
+      validation: {
+        maxWords,
+        mandatory: optional !== 'true',
+      },
+    });
+
+    return {
+      data: 'QUESTION_SAVED',
+      sessionId,
+    };
+  };
+
+  const onSuccessRedirectHref = (response: {
+    redirectQuestionType?: ResponseTypeEnum;
+  }) => {
+    if (response.redirectQuestionType) {
+      return getRedirect(
+        response.redirectQuestionType,
+        applicationId,
+        sectionId
+      );
     }
-  }
+    return `/build-application/${applicationId}/${sectionId}`;
+  };
+
+  const fetchPageData = async (sessionCookie: string) => {
+    try {
+      const applicationFormSummary = await getApplicationFormSummary(
+        applicationId,
+        sessionCookie
+      );
+      const responseType = await getValueFromSession(
+        'newQuestion',
+        'responseType',
+        sessionCookie
+      );
 
-  const sectionName = applicationFormSummary.sections.find(
-    (section) => section.sectionId === sectionId
-  )?.sectionTitle;
+      const sectionName = applicationFormSummary.sections.find(
+        (section) => section.sectionId === sectionId
+      )?.sectionTitle;
 
-  const { publicRuntimeConfig } = getConfig();
-  return {
-    props: {
-      sectionName: sectionName,
-      defaultRadio: selectedRadio,
-      backButtonHref: `/build-application/${applicationId}/${sectionId}/question-content`,
-      fieldErrors: fieldErrors,
-      formAction: `${publicRuntimeConfig.SUB_PATH}/build-application/${applicationId}/${sectionId}/question-type`,
-      csrfToken: res.getHeader('x-csrf-token') as string,
-    },
+      return {
+        sectionName,
+        defaultRadio: getRadio(responseType),
+        backButtonHref: `/build-application/${applicationId}/${sectionId}/question-content`,
+      };
+    } catch (err: unknown) {
+      return questionErrorPageRedirect(applicationId);
+    }
```

Comment:
Can remove try catches - `QuestionPageGetServerSideProps` handles this

Reply:
We want a different error page than the one rendered by QuestionPageGetServerSideProps. Wasn't super keen on adding more and more logic in there if we can avoid it

Reply:
Whats the difference we want in the error page?

Reply:
The href attached to the return anchor tag needs to not be the resolved url. It needs to go to the dashboard, I can get around it by adding a `fetchPageDataErrorHandler` but I can't get a nice solution there for both the get and post scenarios because the error handler seems to be called in. both.

have had a wee look - the only other file which uses `fetchPageDataErrorHandler` is the super admin dashboard which has a handleRequest which wont ever throw ->   `const handleRequest = async (body: SuperAdminDashboardFilterData) => body;` I have got around this issue by just removing the check for it in the POST handler

Reply:
As discussed can get rid of this prop, as `fetchPageData` can return redirects and in the one place we need some special try catch behaviour we can just shift it into that function